kolejka/judge/systems/local.py

```diff
@@ -223,6 +276,10 @@ def execute_safe_command(self, command, stdin_path, stdout_path, stdout_append,
         resources = self.get_resources(limits)
         #resources[resource.RLIMIT_NPROC] = (1,1) #This is a very bad idea, read notes in man execv on EAGAIN
 
+        memory_reservation_thread = None
+        if limits.gpu_memory:
+            memory_reservation_thread = threading.Thread(target=memory_reservation, args=(limits.gpu_memory,))
+            memory_reservation_thread.start()
```

Comment:
Unlikely (?) race condition with process. We should wait with process_start until memory reservation is acknowledged?

Reply:
Good point. At first, I was struggling with the termination of a thread so just copy-pasting logic from monitor process and I forgot to rethink the overall situation. I've transformed the memory reservation thread into process (😢) due to the ability to explicitly terminate right after the main command.